src/Compilers/CSharp/Portable/Compilation/BuiltInOperators.cs

```diff
@@ -741,19 +742,24 @@ internal static bool IsValidObjectEquality(Conversions Conversions, TypeSymbol l
             }
 
             var leftIsReferenceType = ((object)leftType != null) && leftType.IsReferenceType;
-            if (!leftIsReferenceType && !leftIsNull)
+            if (!leftIsReferenceType && !leftIsNull && !leftIsDefault)
             {
                 return false;
             }
 
             var rightIsReferenceType = ((object)rightType != null) && rightType.IsReferenceType;
-            if (!rightIsReferenceType && !rightIsNull)
+            if (!rightIsReferenceType && !rightIsNull && !rightIsDefault)
             {
                 return false;
             }
 
-            // If at least one side is null then clearly a conversion exists.
-            if (leftIsNull || rightIsNull)
+            if (leftIsDefault && rightIsDefault)
+            {
+                return false;
+            }
+
+            // If at least one side is null or default then clearly a conversion exists.
```

Comment:
>// If at least one side is null or default then clearly a conversion exists. [](start = 12, length = 76)

Is there a benefit in allowing ```null == default```? Is it always true?

Reply:
Good catch. Thanks
I'll keep as disallowed as it doesn't seem useful.

---
In reply to: [328337637](https://github.com/dotnet/roslyn/pull/38853#discussion_r328337637) [](ancestors = 328337637)